### DISCUSSION TITLE
How can I detect where I'm coming from?

### DISCUSSION THREAD

**Discussion by widert** [zwave-js/node-red-contrib-zwave-js#261] [Q&A] [answered]
I'm trying to use getValue from a Thermostat using ValueAPI in a function, and everything works fine.

However, if I have several z-wave device nodes all of them are "trigger", i.e. if I click on the inject of the upper flow as shown on image below, the lower is also triggered, resulting in two debug messages.

How can I detect where I'm coming from (e.g. what injection button I am clicking)?

![Capture](https://user-images.githubusercontent.com/29115503/202313715-2f29dd86-8da8-415b-9c04-d053381b6a3e.PNG)

**Comment by marcus-j-davies** [comment, marked as answer]
Hi @widert,

This is something that will be improved upon in V9.

However, in the meantime, enable the 'Isolated get value' option in each of your device nodes.

This stops other device nodes from being triggered.

I.e the result is only passed to the device node that sent the get value request.
